rust/src/router/producer.rs

```diff
@@ -179,28 +240,72 @@ pub struct ProducerStat {
     pub ssrc: u32,
     pub rtx_ssrc: Option<u32>,
     pub rid: Option<String>,
-    pub kind: String,
+    pub kind: MediaKind,
     pub mime_type: MimeType,
-    pub packets_lost: u32,
+    pub packets_lost: u64,
     pub fraction_lost: u8,
-    pub packets_discarded: usize,
-    pub packets_retransmitted: usize,
-    pub packets_repaired: usize,
-    pub nack_count: usize,
-    pub nack_packet_count: usize,
-    pub pli_count: usize,
-    pub fir_count: usize,
+    pub packets_discarded: u64,
+    pub packets_retransmitted: u64,
+    pub packets_repaired: u64,
+    pub nack_count: u64,
+    pub nack_packet_count: u64,
+    pub pli_count: u64,
+    pub fir_count: u64,
     pub score: u8,
-    pub packet_count: usize,
-    pub byte_count: usize,
+    pub packet_count: u64,
+    pub byte_count: u64,
     pub bitrate: u32,
     pub round_trip_time: Option<f32>,
-    pub rtx_packets_discarded: Option<u32>,
+    pub rtx_packets_discarded: Option<u64>,
     // RtpStreamRecv specific.
     pub jitter: u32,
-    pub bitrate_by_layer: Option<HashedMap<String, u32>>,
+    pub bitrate_by_layer: Vec<BitrateByLayer>,
 }
 
+impl ProducerStat {
+    pub(crate) fn from_fbs(stats: &rtp_stream::Stats) -> Self {
+        let rtp_stream::StatsData::RecvStats(ref stats) = stats.data else {
+            panic!("Wrong message from worker");
+        };
+
+        let rtp_stream::StatsData::BaseStats(ref base) = stats.base.data else {
+            panic!("Wrong message from worker");
+        };
+
+        Self {
+            timestamp: base.timestamp,
+            ssrc: base.ssrc,
+            rtx_ssrc: base.rtx_ssrc,
+            rid: base.rid.clone(),
+            kind: MediaKind::from_fbs(base.kind),
+            mime_type: base.mime_type.to_string().parse().unwrap(),
```

Comment:
cc @nazar-pc. I see I'm abusing calling to `unwrap()`. I'll review FBS transform methods and probably need to make them return most of the times a Result.